### DISCUSSION TITLE
Early trial stop based on some time limit

### DISCUSSION THREAD

**Discussion by FinderSerg** [microsoft/nni#3740] [Ideas]
**What would you like to be added**:
Max time a trial can run after which it would be terminated. 
```yml
"maxExperimentDuration": "86400s",
"maxTrialDuration": "600s",
```  

**Why is this needed**:
Some of the parameters sets takes a lot more time to evaluate than other and it would be nice to have possibility to terminate them based on some time limit to check more other sets in given time.

**Without this feature, how does current nni work**：
It takes a lot of time for some trials or even stuck on them until trials is manually terminated.

**Components that may involve changes**:
Do not know

**Brief description of your proposal if any**:
Add field `maxTrialDuration` to config

**Comment by scarlett2018** [comment]
@FinderSerg - thanks for raising this new feature request. I'm moving this to discussion so that the issue is also visible to others to vote. 

Guys - please use **thumb up** to vote for adding this feature.
